### PR TITLE
Add machine deployment restart endpoint

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
API part of https://github.com/kubermatic/dashboard/issues/3440

**Does this PR introduce a user-facing change?**:

```release-note
Added endpoint to restart machine deployments.
```

### CODE REVIEW

File: pkg/handler/common/machine.go [hunk 1]
```diff
@@ -521,6 +522,35 @@ func PatchMachineDeployment(ctx context.Context, userInfoGetter provider.UserInf
...
+
+	machineDeployment := &clusterv1alpha1.MachineDeployment{}
+	if err := client.Get(ctx, types.NamespacedName{Namespace: metav1.NamespaceSystem, Name: machineDeploymentID}, machineDeployment); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	if machineDeployment.Annotations == nil {
+		machineDeployment.Annotations = map[string]string{}
+	}
+	machineDeployment.Annotations["forceRestart"] = strconv.FormatInt(time.Now().UnixNano(), 10)
```
Comment (zreigz) [inline pkg/handler/common/machine.go]: do we have const for this key somewhere?

Comment (maciaszczykm) [reply pkg/handler/common/machine.go]: No, there isn't any. I'll add a new one.

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**